pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/LiteralTransformFunction.java

```diff
@@ -45,30 +46,32 @@ public class LiteralTransformFunction implements TransformFunction {
   private final long _longLiteral;
   private final float _floatLiteral;
   private final double _doubleLiteral;
+  private final BigDecimal _bigDecimalLiteral;
 
   // literals may be shared but values are intentionally not volatile as assignment races are benign
   private int[] _intResult;
   private long[] _longResult;
   private float[] _floatResult;
   private double[] _doubleResult;
+  private BigDecimal[] _bigDecimalResult;
   private String[] _stringResult;
   private byte[][] _bytesResult;
 
   public LiteralTransformFunction(String literal) {
     _literal = literal;
     _dataType = inferLiteralDataType(literal);
-    if (_dataType.isNumeric()) {
-      BigDecimal bigDecimal = new BigDecimal(_literal);
-      _intLiteral = bigDecimal.intValue();
-      _longLiteral = bigDecimal.longValue();
-      _floatLiteral = bigDecimal.floatValue();
-      _doubleLiteral = bigDecimal.doubleValue();
+    if (_dataType == DataType.TIMESTAMP) {
+      _bigDecimalLiteral = BigDecimal.valueOf(Timestamp.valueOf(literal).getTime());
```

Comment:
_bigDecimalLiteral is used to fill all other literal values including: _longLiteral.

Reply:
inferLiteralDataType() can return DataType.TIMESTAMP. Once _bigDecimalLiteral is set, other literal values including _longLiteral is set.

Reply:
I see. Let's add some comments explaining the behavior because it can cause confusion. Also, we should handle `BOOLEAN` as well (true should map to 1)